app/main/command-line.js

```diff
@@ -0,0 +1,82 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+/**
+ * This module handles opening a URL when starting Tofino with command line args, or
+ * opening a URL when Tofino is the default browser, or when dragging a file to be loaded
+ * via Tofino. Most of this logic is taken from Brave's implementation.
+ * https://github.com/brave/browser-laptop/blob/master/app/cmdLine.js
+ */
+import electron from 'electron';
+import { parse } from 'url';
+import * as BW from './browser-window';
+import * as protocols from './protocols';
+
+const { app } = electron;
+const isDarwin = process.platform === 'darwin';
+
+// Checks an array of arguments if it can find a url
+const getUrlFromCommandLine = argv => {
+  if (argv) {
+    if (argv.length === 2 && !argv[1].startsWith('-')) {
+      const parsedUrl = parse(argv[1]);
+      if (protocols.DEFAULT_PROTOCOLS.includes(parsedUrl.protocol)) {
+        return argv[1];
+      }
+    }
+    const index = argv.indexOf('--');
+    if (index !== -1 && index + 1 < argv.length && !argv[index + 1].startsWith('-')) {
+      const parsedUrl = parse(argv[index + 1]);
+      if (protocols.DEFAULT_PROTOCOLS.includes(parsedUrl.protocol)) {
+        return argv[index + 1];
+      }
+    }
+  }
+  return undefined;
+};
+
+// For macOS, there are events like open-url instead
+if (!isDarwin) {
+  const openUrl = getUrlFromCommandLine(process.argv);
```

Comment:
Note that the yargs stuff has special handling for electron where sometimes you have an extra random argument.